flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -245,14 +235,15 @@ public int getKeyGroupPrefixBytes() {
 		// hold the db lock while operation on the db to guard us against async db disposal
 		synchronized (asyncSnapshotLock) {
 
-			if (kvStateInformation.isEmpty()) {
-				LOG.info("Asynchronous RocksDB snapshot performed on empty keyed state at " + timestamp +
-						" . Returning null.");
+			if (db != null) {
 
-				return new DoneFuture<>(null);
-			}
+				if (kvStateInformation.isEmpty()) {
+					LOG.info("Asynchronous RocksDB snapshot performed on empty keyed state at " + timestamp +
```

Comment:
Wondering whether we should log this on debug. As a user, I don't know how useful this is for me when I see this message in my logs.